src/app/components/named-entity-ref/named-entity-ref.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, Input } from '@angular/core';
+import { of } from 'rxjs';
+import { map, share } from 'rxjs/operators';
+import { NamedEntityRefData } from '../../models/evt-models';
+import { register } from '../../services/component-register.service';
+import { NamedEntitiesParserService } from '../../services/xml-parsers/named-entities-parser.service';
+
+@Component({
+  selector: 'evt-named-entity-ref',
+  templateUrl: './named-entity-ref.component.html',
+  styleUrls: ['./named-entity-ref.component.scss'],
+})
+@register
+export class NamedEntityRefComponent {
+  @Input() data: NamedEntityRefData;
+
+  entity$ = this.neParserService.namedEntities$.pipe(
+    share(),
+    map(ne => ne.all.entities.find(e => e.id === this.data.entityId) || 'notFound'),
```

Comment:
`'notFound'` is not af `NamedEntity` type!

Reply:
I need a way to distinguish among three different statuses: 
- XML Not yet parsed => `undefined`
- XML Parsed and entity found => `NamedEntity` object
- XML Parsed but entity not found => `'notFound'` string

Do you have a better implementation to suggest?

Reply:
You need to change the entities parser, let's leave it as it is but it should be changed